internal/Yesod/Session/Persist/SessionKey.hs

```diff
@@ -0,0 +1,51 @@
+module Yesod.Session.Persist.SessionKey
+  ( SessionKey (..)
+  , SessionKeyManager (..)
+  , makeSessionKeyManager
+  , sessionKeyToCookieValue
+  , sessionKeyFromCookieValue
+  )
+where
+
+import Yesod.Session.Persist.Prelude
+
+import Data.ByteString.Base64.URL qualified as B64URL
+import Data.ByteString.Char8 qualified as BS8
+import Data.Text qualified as T
+
+data SessionKeyManager m = SessionKeyManager
+  { new :: m SessionKey
+  -- ^ Generate a new session key
+  --
+  -- In a production setting, it is critical that this action be
+  -- thread-safe and produce a securely random result.
+  , check :: SessionKey -> Bool
+  -- ^ Validate that a text is something that plausibly could have
+  --   been generated by 'new'.
+  }
+
+-- | Secret value that is sent to and subsequently furnished by
+--   the client to identify the session
+newtype SessionKey = SessionKey {text :: Text}
+  deriving newtype (Eq, Ord, Show)
+
+makeSessionKeyManager :: Monad m => Randomization m -> SessionKeyManager m
+makeSessionKeyManager (Randomization generateRandomBytes) =
+  let
+    new = SessionKey . B64URL.encodeBase64 <$> generateRandomBytes 18
+    check (SessionKey text) =
+      T.length text
+        == 24
+        && either
+          (const False)
+          ((== 18) . BS8.length)
```

Comment:
Is there some way to demystify these magic numbers?

Reply:
Defined constants and added comments